src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSyncController.scala

```diff
@@ -0,0 +1,171 @@
+package io.iohk.ethereum.blockchain.sync
+
+import akka.actor._
+import io.iohk.ethereum.blockchain.sync.FastSyncController.StartRegularSync
+import io.iohk.ethereum.db.storage.AppStateStorage
+import io.iohk.ethereum.domain.{Block, BlockHeader, Blockchain}
+import io.iohk.ethereum.network.PeerActor.{BroadcastBlocks, MessageReceived, SendMessage, Subscribe}
+import io.iohk.ethereum.network.p2p.messages.CommonMessages.{NewBlock, Status}
+import io.iohk.ethereum.network.p2p.messages.PV62._
+import io.iohk.ethereum.network.p2p.validators.BlockValidator.BlockError
+import io.iohk.ethereum.utils.Config
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.util.Try
+
+trait RegularSyncController {
+  self: Actor with ActorLogging with BlacklistSupport =>
+
+  val appStateStorage: AppStateStorage
+  val blockchain: Blockchain
+  val blockValidator: (BlockHeader, BlockBody) => Either[BlockError, Block]
+  def peersToDownloadFrom: Map[ActorRef, Status]
+
+  private var headers: Seq[BlockHeader] = Seq.empty
```

Comment:
can we name that more precisely? are these all headers we have? all headers downloaded during regular sync? headers downloaded in some time period?